src/Core/View.js

```diff
@@ -100,8 +100,12 @@ const _syncThreejsLayer = function _syncThreejsLayer(layer, view) {
 function _preprocessLayer(view, layer, provider) {
     if (!(layer instanceof Layer) && !(layer instanceof GeometryLayer)) {
         const nlayer = new Layer(layer.id);
-        delete layer.id;
+        // nlayer.id is read-only so delete it from layer before Object.assign
+        const tmp = layer;
+        delete tmp.id;
         layer = Object.assign(nlayer, layer);
+        // restore layer.id in user provider layer object
+        tmp.id = layer.id;
```

Comment:
We have a to-be-reactivated linter rule that forbids modifying parameters. Guess it'll be useful (because bug like that are hard to find).

Also, I'm wondering if all this part shouldn't be in Layer constructor? Then we won't even need to do this stunt: we'll just copy layer fields before creating the read-only property. WDYT?